### PR TITLE
Prod

### PR DESCRIPTION
**Pull request by javorskyp**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -47,16 +67,28 @@ class App extends Component {
...
-        <Header onSearch={term => this.searchHandler(term)}/>
-        <Menu />
-        {this.state.loading 
-          ? <LoadingIcon />
-          : <Motorcycles motorcycles={this.state.motorcycles} />
-        }
-      </div>
+            <Layout
+              header={<Header>
+                        <Searchbar onSearch={term => this.searchHandler(term)} />
```
Comment (Aciderr) [inline src/App.js]: You should be able to use onSearch={this.searchHandler} - check this out

File: src/App.js [hunk 2]
```diff
@@ -47,16 +67,28 @@ class App extends Component {
     }, 1000);
   }
 
+  changeTheme = () => {
+    const newTheme = this.state.theme === 'primary' ? 'danger' : 'primary';
+    this.setState({theme:newTheme})
+  }
+
   render() {
```
Comment (Aciderr) [inline src/App.js:90]: Please use: 
const { theme, motorcycles } = this.state 
just after render() {
And then in template use can use theme and motorcycles consts.
Using this.state.theme is quite repetitive.

File: src/components/Footer/Footer.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const Footer = (props) =>  (
+    <div className={`text-center m-5-${props.theme}`}>Wypożyczalnia FreeRide 2021</div>
+);
```
Comment (Aciderr) [inline src/components/Footer/Footer.js]: Please use dynamic date something like new Date().getFullYear()

Comment (Aciderr) [reply src/components/Footer/Footer.js]: You haven't specified PropTypes for this and some others components

Comment (javorskyp) [reply src/components/Footer/Footer.js]: Ok, tomorrow I should change date 


yes, i know about props, later i do sth with this

File: src/App.js (hunk 1 above)
Comment (javorskyp) [reply src/App.js]: Done

File: src/components/Footer/Footer.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 import ThemeContext from '../../context/themeContext';
 
-const Footer = (props) => {
+export default function Footer() {
   const theme = useContext(ThemeContext);
+  const [date , setDate] = useState();
+  const getYear = () =>  setDate(new Date().getFullYear())
+  useEffect(() => {
+      getYear();
+  }, [])
```
Comment (Aciderr) [inline src/components/Footer/Footer.js]: This should be written as:
const date = new Date().getFullYear();`
without any state, effect - will be simpler.

File: src/hoc/withMousePosition.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      x: 0,
+      y: 0
+    };
+
+    componentDidMount() {
+      document.body.addEventListener(
+          'mousemove', 
+          this.updateMousePostion.bind(this)
+        );
+    }
```
Comment (Aciderr) [inline src/hoc/withMousePosition.js:15]: In componentWillUnmount You need to use removeEventListener

File: src/hoc/withMousePosition.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    updateMousePostion(e) {
+      this.setState({
+        x: e.pageX,
+        y: e.pageY
+      });
+    }
```
Comment (Aciderr) [inline src/hoc/withMousePosition.js:29]: Try to change this function to an arrow function so the .bind(this) could be removed.
In my current company, we have an approach that we use only arrow functions. You can also try to keep that convention.

**Review by Aciderr — APPROVED** [review]

**Pull request merged**